app/init/resources.php

```diff
@@ -691,6 +693,75 @@ function getDevice(string $root, string $connection = ''): Device
     return $register->get('geodb');
 }, ['register']);
 
+App::setResource('geoRecord', function (Reader $geodb, Request $request, Locale $locale) {
+    $ip = $request->getIp();
+
+    if (!filter_var($ip, FILTER_VALIDATE_IP)) {
+        Console::warning("Invalid IP address: {$ip}");
+        $ip = '0.0.0.0'; // Use fallback IP
+    }
+    $eu = Config::getParam('locale-eu');
+    $currencies = Config::getParam('locale-currencies');
+
+    $client = new Client();
+    $client->addHeader('Authorization', 'Bearer ' . System::getEnv('_APP_GEO_SECRET'));
+
+    $client->setBaseUrl('http://appwrite-geo/v1');
+
+    $record = null;
+    try {
+        $record = $client->fetch("/ips/{$ip}", Client::METHOD_GET);
+        if ($record->getStatusCode() !== 200) {
+            $error = $record->json();
+            throw new Exception(Exception::GENERAL_SERVER_ERROR, $error['message'] ?? 'Failed to fetch geo data');
+        }
+        $record = $record->json();
+    } catch (Throwable $th) {
+        Console::error($th->getMessage());
+        Console::error($th->getTraceAsString());
+        $record = null;
+    }
+
+    if (empty($record)) {
+        // Fallback
+        $dbRecord = $geodb->get($ip);
+        if ($dbRecord) {
+            $record = [];
+            $record['countryCode'] = $dbRecord['country']['iso_code'] ?? '--';
+            $record['country'] = $dbRecord['country']['names'] ?? [];
+            $record['continent'] = $dbRecord['continent']['names'] ?? [];
+            $record['continentCode'] = $dbRecord['continent']['code'] ?? '--';
+        }
+    }
+
+    $output = [];
```

Comment:
I think we should use Document here, to be consistent with other resources

Reply:
right, this makes sense.